rosbag2_py/test/test_sequential_writer.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+import shutil
+
+from common import get_rosbag_options
+from rclpy.serialization import deserialize_message, serialize_message
+import rosbag2_py
+from rosidl_runtime_py.utilities import get_message
+from std_msgs.msg import String
+
+
+def create_topic(writer, topic_name, topic_type, serialization_format='cdr'):
+    """
+    Create a new topic.
+
+    :param writer: writer instance
+    :param topic_name:
+    :param topic_type:
+    :param serialization_format:
+    :return:
+    """
+    topic_name = topic_name
+    topic = rosbag2_py.TopicMetadata()
+    topic.name = topic_name
+    topic.serialization_format = serialization_format
+    topic.type = topic_type
+
+    writer.create_topic(topic)
+
+
+def test_sequential_writer():
+    """
+    Test for sequential writer.
+
+    :return:
+    """
+    bag_path = str(Path(__file__).parent.parent / 'resources' / 'tmp_write_test')
```

Comment:
I think it would be better if we didn't leave test artifacts lingering in the source tree. Also, for some reason it doesn't look like the `shutil.rmtree` call is working on my machine.

I think it's better to use a `tempfile.TemporaryDirectory` context for this whole test, e.g.

```python
    with tempfile.TemporaryDirectory() as tmpdir:
        bag_path = os.path.join(tmpdir, 'tmp_write_test')

        # rest of the test here
```

Then we don't need to worry about checking for the existence of `bag_path` or cleaning up the temporary directory afterwards.

Reply:
Nice. `shutil.rmtree` not working explains why I was getting this other error on CI when I reran without changing anything but a printout:

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=11653)](http://ci.ros2.org/job/ci_linux/11653/)
```
>       writer.open(storage_options, converter_options)
E       RuntimeError: Database direcotory already exists (/home/jenkins-agent/workspace/ci_linux/ws/src/mabelzhang/rosbag2/rosbag2_py/resources/tmp_write_test), can't overwrite existing database
```

I will try the `tempfile` and rerun CI.